content/en/docs/concepts/storage/storage-capacity.md

```diff
@@ -0,0 +1,116 @@
+driver](/docs/concepts/storage/volumes/#csi) deployment also has to
+support it. Please refer to the driver's documentation for
+details. The feature is not supported for non-CSI storage systems.
+
+Without this support, there will be no information about storage
+capacity available through the driver and the scheduler will schedule
+Pods with volumes provided by the driver without looking for capacity
+information.
+
+## API
+
+There are two API extensions for this feature:
+- [`CSIStorageCapacity` objects](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#csistoragecapacity-v1alpha1-storage-k8s-io):
+  these get produced by a CSI driver in the namespace
+  where the driver is installed. Each object contains capacity
+  information for one storage class and defines which nodes have
+  access to that storage.
+- [The `CSIDriverSpec.StorageCapacity` field](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#csidriverspec-v1-storage-k8s-io):
+  when set to `true`, the Kubernetes scheduler will consider storage
+  capacity for volumes that use the CSI driver.
+
+## Scheduling
+
+Storage capacity information is used by the Kubernetes scheduler if:
+- the `CSIStorageCapacity` feature gate is true,
+- a Pod uses a volume that has not been created yet,
+- that volume uses a storage class which references a CSI driver and
+  uses [`WaitForFirstConsumer` volume binding
+  mode](/docs/concepts/storage/storage-classes/#volume-binding-mode),
+  and
+- the `CSIDriver` object for the driver has `StorageCapacity` set to
+  true.
+
+In that case, the scheduler only considers nodes for the Pod which
+have enough storage available to them. This check is very
+simplistic and only compares the size of the volume against the
+capacity listed in `CSIStorageCapacity` objects with a topology that
+includes the node.
+
+For volumes with `Immediate` volume binding mode, the storage driver
+decides where to create the volume, independently of Pods that will
+use the volume. The scheduler then schedules Pods onto nodes where the
+volume is available after the volume has been created.
+
+For [CSI ephemeral volumes](/docs/concepts/storage/volumes/#csi),
+scheduling always happens without considering storage capacity. This
+is based on the assumption that this volume type is only used by
+special CSI drivers which are local to a node and do not need
+significant resources there.
+
+## Rescheduling
+
+When a node has been selected for a Pod with `WaitForFirstConsumer`
+volumes, that decision is still tentative. The next step is that the
+CSI storage driver gets asked to create the volume with a hint that the
+volume is supposed to be available on the selected node.
+
+Because Kubernetes might have chosen a node based on out-dated
+capacity information, it is possible that the volume cannot really be
+created. The node selection is then reset and the Kubernetes scheduler
+tries again to find a node for the Pod.
+
+## {{% heading "whatsnext" %}}
+
```

Comment:
Is it useful to link to https://kubernetes.io/docs/reference/scheduling/ or some other scheduling-relevant pages?

Reply:
Yes. I think the concept section may be more useful here, so I added:
`Learn about [Kubernetes Scheduler](/docs/concepts/scheduling-eviction/kube-scheduler/)`